lib/Optimizer/Transforms/DecompositionPatterns.cpp

```diff
@@ -137,6 +137,22 @@ struct HToPhasedRx : public OpRewritePattern<quake::HOp> {
   }
 };
 
+// Naive mapping of R1 to Rz, ignoring the global phase.
+// This is only expected to work with full inlining and
+// quake apply specialization.
+struct R1ToRz : public OpRewritePattern<quake::R1Op> {
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(quake::R1Op r1Op,
+                                PatternRewriter &rewriter) const override {
+    if (!r1Op.getControls().empty())
+      return failure();
+
+    rewriter.replaceOpWithNewOp<quake::RzOp>(
+        r1Op, r1Op.getParameters(), r1Op.getControls(), r1Op.getTargets());
```

Comment:
You will also have to pass the adjoint attribute to the constructor, otherwise if `r1` is in adjoint form we will lose this information.

Reply:
Done.